### PR TITLE
Add missing `GoogleProductResponse` fields

### PR DESCRIPTION
**Pull request by candidcold**
This PR adds fields that previously weren't being returned by Register:

* `subscriptionPeriod`
* `freeTrialPeriod`
* `introductoryPrice`
* `introductoryPriceMicros`
* `introductoryPricePeriod`
* `introductoryPriceCycles`

### CODE REVIEW

File: lib/src/androidTestDebug/java/com/nytimes/android/external/registerlib/GoogleProductResponseTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import org.json.JSONObject;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class GoogleProductResponseTest {
+
+
+    @Test
+    public void testGoodString(){
```
Comment (wdziemia) [inline lib/src/androidTestDebug/java/com/nytimes/android/external/registerlib/GoogleProductResponseTest.java]: Can we remove refs to NYT and keep it generic (productId, subscriptionPeriod)

File: lib/src/androidTestDebug/java/com/nytimes/android/external/registerlib/GoogleProductResponseTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        String json2 = "{ \"description\": " +
```
Comment (wdziemia) [inline lib/src/androidTestDebug/java/com/nytimes/android/external/registerlib/GoogleProductResponseTest.java]: Can we remove refs to NYT and keep it generic (productId, subscriptionPeriod)

File: lib/src/androidTestDebug/java/com/nytimes/android/external/registerlib/GoogleProductResponseTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        String bad = "{ \"description\": " +
+                "\"Basic access features, plus NYT Crossword, Cooking, and one bonus \\nsubscription.\", " +
+                "\"freeTrialPeriod\": \"P1W\", \"price\": \"$24.99\", \"price_amount_micros\": 24990000, " +
+                "\"price_currency_code\": \"USD\", " +
+                "\"productId\": \"com.nytimes.googleplay.bundle.maxadacr.v1.monthly\", " +
+                "\"subscriptionPeriod\": \"P1M\", " +
+                "\"title\": \"NYT All Access (NYTimes - Latest News)\", \"type\": \"subs\" }";
+
+        GoogleProductResponse response = GoogleProductResponse.fromJson(bad);
+        assertNotNull(response);
```
Comment (wdziemia) [inline lib/src/androidTestDebug/java/com/nytimes/android/external/registerlib/GoogleProductResponseTest.java]: If this is bad JSON, shouldnt the response be null?

File: lib/src/androidTestDebug/java/com/nytimes/android/external/registerlib/GoogleProductResponseTest.java (hunk 2 above)
Comment (wdziemia) [reply lib/src/androidTestDebug/java/com/nytimes/android/external/registerlib/GoogleProductResponseTest.java]: Also variable name from `json2` to something more descriptive?

File: lib/src/androidTestDebug/java/com/nytimes/android/external/registerlib/GoogleProductResponseTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package com.nytimes.android.external.registerlib;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class GoogleProductResponseTest {
```
Comment (wdziemia) [inline lib/src/androidTestDebug/java/com/nytimes/android/external/registerlib/GoogleProductResponseTest.java]: Can we use assetJ for the assertions?
`assertThat(CODE.METHOD()).isNull()`
`assertThat(CODE.METHOD()).isNotNull()`

File: lib/src/androidTestDebug/java/com/nytimes/android/external/registerlib/GoogleProductResponseTest.java (hunk 3 above)
Comment (cybo42) [reply lib/src/androidTestDebug/java/com/nytimes/android/external/registerlib/GoogleProductResponseTest.java]: Previously this was causing an exception to be thrown and not deserializing the data.  
Should maybe make the test name more descriptive of that. Not so much bad, but missing some optional fields.

File: lib/src/main/java/com/nytimes/android/external/registerlib/GoogleProductResponse.java [hunk 5]
```diff
@@ -20,14 +20,26 @@
...
+    private static final String FLD_INTRODUCTORY_PRICE_PERIOD = "introductoryPricePeriod";
+    private static final String FLD_INTRODUCTORY_PRICE_CYCLES = "introductoryPriceCycles";
 
     String productId;
     String itemType;
     String price;
     String title;
     String description;
-    int priceAmountMicros;
+    Integer priceAmountMicros;
```
Comment (goldstei) [inline lib/src/main/java/com/nytimes/android/external/registerlib/GoogleProductResponse.java]: Micros fields should be `Long`

**Comment by wdziemia** [comment]
Should this be part of 1.0 Milestone?

**Comment by candidcold** [comment]
Most likely yes, but things might be weird given the "shadow releases" we made, so not sure if this is ready to merge

@goldstei if you ever get a chance could you shed some light

**Comment by goldstei** [comment]
it does need to be part of 1.0.   NYTimes core app branch that is ready for this change

**Comment by wdziemia** [comment]
@goldstei @candidcold Can we merge?

**Comment by wdziemia** [comment]
#70 is blocked by this PR

**Pull request merged**